Add support for EST Clients to advertise support for automated cert p… (#7)
AMWA-TV/nmos-certificate-provisioning: docs/1.0. Certificate Provisioning.md

An EST Client MAY ignore all timestamps in the certificate validity periods during the certificate provisioning process if it does not accurately know the current time, see [Bootstrapping Remote Secure Key Infrastructures - Section 2.6.1](https://tools.ietf.org/html/draft-ietf-anima-bootstrapping-keyinfra-41#section-2.6.1) for more information.

If an EST Client implements the NMOS IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'tags' attribute of the Node API's `/self` endpoint. Control interfaces can identify all Nodes which support BCP-003-03 by a key 'cert'. The supported version of this BCP is defined by the 'versions' list and if EST is currently active by the 'enabled' boolean.

**Example:** The 'tags' attribute of a NMOS Node
```json
...
"tags": {

>   "cert": {

Reviewer: ```suggestion
  "urn:x-nmos:tag:cert": {
```

Matching the precedent in https://github.com/AMWA-TV/nmos-parameter-registers/blob/master/tags/grouphint.md a little more closely. The version could potentially make its way into the tag too.

An EST Client MAY ignore all timestamps in the certificate validity periods during the certificate provisioning process if it does not accurately know the current time, see [Bootstrapping Remote Secure Key Infrastructures - Section 2.6.1](https://tools.ietf.org/html/draft-ietf-anima-bootstrapping-keyinfra-41#section-2.6.1) for more information.

> If an EST Client implements the NMOS IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'caps' attribute of the Node API. Control interfaces can identify all Nodes which support BCP-003-03 by a key 'cert' and Uniform Resource Name (URN) having the base 'urn:x-nmos:control:cert' followed by a '/' character and the supported version.

Reviewer: ```suggestion
If an EST Client implements the NMOS IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'caps' attribute of the Node API. Control interfaces can identify all Nodes which support BCP-003-03 by a key 'cert' and Uniform Resource Name (URN) having the base 'urn:x-nmos:control:cert' followed by a '/' character and the supported version of this BCP.
```

We probably need a mechanism to cope with a client which supports multiple versions of the BCP. For Device 'controls' the array allows us to have multiple entries with different versioning.

**Example:** The 'tags' attribute of a NMOS Node
```json
...
"tags": {

>   "urn:x-nmos:tag:certprov/v1.0": ["enabled"]

Reviewer 2: :thought_balloon: Might this be reasonable? keep one tag that has a list of supported versions, similar to the api pages?

```suggestion
  "urn:x-nmos:tag:certprov": ["v1.0"]
```

Reviewer 3: I was wondering about this! The only downside I see is that didn't someone ask about distinguishing between support for the spec and that support being currently active/enabled? But I think it's entirely reasonable for the resource to indicate the current configuration, so personally I like this proposal better.

Reviewer 2: > The only downside I see is that didn't someone ask about distinguishing between support for the spec and that support being currently active/enabled?

I would explicitly say listed versions are active, if the tag or any version is missing then it is unsupported/disabled. Another talking point for the next call?

Reviewer 3: Your proposal makes perfect sense to me.

Reviewer: Yes, that makes sense to me too. We just need to make sure the text is clear that this shouldn't be present if the EST support is currently disabled by configuration.

Author: Sounds good, I have updated the docs to reflect this

An EST Client MAY ignore all timestamps in the certificate validity periods during the certificate provisioning process if it does not accurately know the current time, see [Bootstrapping Remote Secure Key Infrastructures - Section 2.6.1](https://tools.ietf.org/html/draft-ietf-anima-bootstrapping-keyinfra-41#section-2.6.1) for more information.

> If an EST Client implements the NMOS IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'tags' attribute of the Node API's `/self` endpoint. Control interfaces can identify all Nodes which support BCP-003-03 by a URN key 'urn:x-nmos:control:cert/', followed by the supported version of this BCP. If EST is currently active is signalled using the 'enabled' boolean.

Reviewer 3: Nitpick. "AMWA IS-04" or just "IS-04", not "NMOS IS-04".

> If an EST Client implements the AMWA IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'tags' attribute of the Node API's `/self` endpoint. Control interfaces can identify all Nodes which support BCP-003-03 by a URN key 'urn:x-nmos:tag:certprov' and a list of enabled BCP-003-03 version. If BCP-003-03 is disabled the list MUST be empty. [NMOS Tags](https://github.com/AMWA-TV/nmos-parameter-registers/tree/master/tags)

Reviewer: ```suggestion
If an EST Client implements the AMWA IS-04 Node API, then the EST Client SHOULD advertise that it supports BCP-003-03 using the 'tags' attribute of the Node API's `/self` endpoint. Control interfaces can identify all Nodes which support BCP-003-03 by a URN key 'urn:x-nmos:tag:certprov' and a list of enabled BCP-003-03 versions. If BCP-003-03 is disabled the list MUST be empty. This tag is defined in the NMOS Parameter Registers under [NMOS Tags](https://github.com/AMWA-TV/nmos-parameter-registers/tree/master/tags).
```
